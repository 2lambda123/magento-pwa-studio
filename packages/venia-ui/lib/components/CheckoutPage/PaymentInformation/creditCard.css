.root {
}

.credit_card_root {
    visibility: visible;
    opacity: 1;
    transition-delay: 64ms;
    transition-duration: 384ms;
    transition-property: opacity, visbility;
    transition-timing-function: var(--venia-anim-standard);
}

.credit_card_root_hidden {
    overflow: hidden;
    visibility: hidden;
    height: 0;
    opacity: 0;
}

.dropin_root {
}

.billing_address_fields_root {
    display: grid;
    gap: 0.5rem;
    padding: 1rem 0rem;
}

.billing_address_fields_root_hidden {
    overflow: hidden;
    visibility: hidden;
    height: 0;
    opacity: 0;
}

.errors_container {
    display: grid;
    gap: 0.5rem;
}

.error {
    color: rgb(var(--venia-error));
}

.first_name,
.last_name {
    grid-column-end: span 1;
}

.country,
.street1,
.street2,
.address_check,
.city,
.state,
.postal_code,
.phone_number {
    grid-column-end: span 2;
}

/*
 * Mobile-specific styles.
 */

@media (max-width: 960px) {
    .first_name,
    .last_name,
    .city,
    .state,
    .postal_code,
    .phone_number,
    .street1,
    .street2,
    .address_check {
        grid-column-end: span 2;
    }
}
