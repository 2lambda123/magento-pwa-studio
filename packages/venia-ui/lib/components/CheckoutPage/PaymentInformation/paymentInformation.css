.root {
    display: grid;
    grid-template-columns: 1fr;
    gap: 2rem;
}

.payment_info_container {
    border: 1px solid rgb(var(--venia-border));
    border-radius: 5px;
}

/*
 * Mobile-specific styles.
 */

@media (max-width: 960px) {
    .container {
        gap: 1rem;
    }
}

@media (max-width: 960px) {
    .heading {
        border-width: 0;
    }
}
