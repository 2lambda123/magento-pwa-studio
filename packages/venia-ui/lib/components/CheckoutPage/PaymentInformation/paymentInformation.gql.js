import gql from 'graphql-tag';

// We disable linting for local fields because there is no way to add them to
// the fetched schema.
// https://github.com/apollographql/eslint-plugin-graphql/issues/99
/* eslint-disable graphql/template-strings */

export const GET_PAYMENT_DETAILS = gql`
    query getSelectedPaymentMethod($cartId: String!) {
        cart(cart_id: $cartId) @connection(key: "Cart") {
            id
            paymentNonce @client
            selectedPaymentMethod: selected_payment_method {
                code
            }
        }
    }
`;

/* eslint-enable graphql/template-strings */

export default {
    queries: {
        getPaymentDetailsQuery: GET_PAYMENT_DETAILS
    },
    mutations: {}
};
