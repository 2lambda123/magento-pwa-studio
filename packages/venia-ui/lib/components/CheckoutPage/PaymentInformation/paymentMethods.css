.root {
    display: grid;
}

.payment_method {
    padding: 1rem;
    border-bottom: 1px solid rgb(var(--venia-border));
}

.payment_method:last-of-type {
    border-bottom-width: 0;
}

.radio_label {
    font-weight: 600;
    width: 6rem;
}
