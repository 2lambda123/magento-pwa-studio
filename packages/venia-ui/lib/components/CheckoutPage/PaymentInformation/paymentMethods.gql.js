import gql from 'graphql-tag';

export const GET_PAYMENT_METHODS = gql`
    query getPaymentMethods($cartId: String!) {
        cart(cart_id: $cartId) @connection(key: "Cart") {
            id
            available_payment_methods {
                code
                title
            }
        }
    }
`;

export default {
    queries: {
        getPaymentMethodsQuery: GET_PAYMENT_METHODS
    },
    mutations: {}
};
