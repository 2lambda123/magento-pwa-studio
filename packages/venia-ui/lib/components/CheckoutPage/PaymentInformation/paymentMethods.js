import React from 'react';
import { shape, string, bool, func } from 'prop-types';
import { RadioGroup } from 'informed';

import Radio from '../../RadioGroup/radio';
import CreditCard from './creditCard';
import { mergeClasses } from '../../../classify';

import defaultClasses from './paymentMethods.css';

const PaymentMethods = props => {
    const {
        classes: propClasses,
        reviewOrderButtonClicked,
        selectedPaymentMethod,
        onPaymentSuccess,
        onPaymentError,
        resetReviewOrderButtonClicked
    } = props;

    const classes = mergeClasses(defaultClasses, propClasses);

    const creditCard =
        selectedPaymentMethod === 'braintree' ? (
            <CreditCard
                shouldSubmit={reviewOrderButtonClicked}
                brainTreeDropinContainerId={
                    'checkout-page-braintree-dropin-container'
                }
                onPaymentSuccess={onPaymentSuccess}
                onPaymentError={onPaymentError}
                resetShouldSubmit={resetReviewOrderButtonClicked}
            />
        ) : null;

    return (
        <div className={classes.root}>
            <RadioGroup field="selectedPaymentMethod" initialValue="braintree">
                <div className={classes.payment_method}>
                    <Radio
                        key={'braintree'}
                        label={'Credit Card'}
                        value={'braintree'}
                        classes={{
                            label: classes.radio_label
                        }}
                        checked={selectedPaymentMethod === 'braintree'}
                    />
                    {creditCard}
                </div>
            </RadioGroup>
        </div>
    );
};

export default PaymentMethods;

PaymentMethods.propTypes = {
    classes: shape({
        root: string,
        payment_method: string,
        radio_label: string
    }),
    reviewOrderButtonClicked: bool,
    selectedPaymentMethod: string,
    onPaymentSuccess: func,
    onPaymentError: func,
    resetReviewOrderButtonClicked: func
};
