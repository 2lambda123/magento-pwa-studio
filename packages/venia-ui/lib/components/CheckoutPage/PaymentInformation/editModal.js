import React from 'react';
import { func, shape, string } from 'prop-types';
import { X as CloseIcon } from 'react-feather';

import { useEditModal } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useEditModal';

import Button from '../../Button';
import Icon from '../../Icon';
import { Modal } from '../../Modal';
import { mergeClasses } from '../../../classify';
import CreditCard from './creditCard';

import editModalOperations from './editModal.gql';

import defaultClasses from './editModal.css';

const EditModal = props => {
    const { classes: propClasses, onClose } = props;

    const classes = mergeClasses(defaultClasses, propClasses);

    const talonProps = useEditModal({ onClose, ...editModalOperations });

    const {
        selectedPaymentMethod,
        isLoading,
        handleUpdate,
        handleClose,
        handlePaymentSuccess,
        handleDropinReady,
        updateButtonClicked,
        resetUpdateButtonClicked,
        handlePaymentError
    } = talonProps;

    const actionButtons = !isLoading ? (
        <div className={classes.actions_container}>
            <Button
                className={classes.cancel_button}
                onClick={handleClose}
                priority="normal"
                disabled={updateButtonClicked}
            >
                {'Cancel'}
            </Button>
            <Button
                className={classes.update_button}
                onClick={handleUpdate}
                priority="high"
                disabled={updateButtonClicked}
            >
                {'Update'}
            </Button>
        </div>
    ) : null;

    const paymentMethod =
        selectedPaymentMethod === 'braintree' ? (
            <div className={classes.body}>
                <CreditCard
                    isHidden={false}
                    shouldSubmit={updateButtonClicked}
                    resetShouldSubmit={resetUpdateButtonClicked}
                    onDropinReady={handleDropinReady}
                    onPaymentSuccess={handlePaymentSuccess}
                    onPaymentError={handlePaymentError}
                    brainTreeDropinContainerId={
                        'edit-modal-braintree-dropin-container'
                    }
                />
                {actionButtons}
            </div>
        ) : (
            <div>{`${selectedPaymentMethod} is not supported for editing.`}</div>
        );

    return (
        <Modal>
            <aside className={classes.root_open}>
                <div className={classes.header}>
                    <span className={classes.header_text}>
                        Edit Payment Information
                    </span>
                    <button
                        className={classes.close_button}
                        onClick={handleClose}
                    >
                        <Icon src={CloseIcon} />
                    </button>
                </div>
                {paymentMethod}
            </aside>
        </Modal>
    );
};

export default EditModal;

EditModal.propTypes = {
    classes: shape({
        root: string,
        root_open: string,
        body: string,
        header: string,
        header_text: string,
        actions_container: string,
        cancel_button: string,
        update_button: string,
        close_button: string
    }),
    onClose: func.isRequired
};
