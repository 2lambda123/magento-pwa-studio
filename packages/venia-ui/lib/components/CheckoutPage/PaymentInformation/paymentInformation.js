import React from 'react';
import { shape, func, string, bool } from 'prop-types';

import { usePaymentInformation } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation';

import PaymentMethods from './paymentMethods';
import Summary from './summary';
import { mergeClasses } from '../../../classify';
import EditModal from './editModal';

import paymentInformationOperations from './paymentInformation.gql';

import defaultClasses from './paymentInformation.css';

const PaymentInformation = props => {
    const {
        onSave,
        classes: propClasses,
        reviewOrderButtonClicked,
        resetReviewOrderButtonClicked,
        isMobile
    } = props;

    const classes = mergeClasses(defaultClasses, propClasses);

    const talonProps = usePaymentInformation({
        ...paymentInformationOperations,
        resetReviewOrderButtonClicked,
        onSave
    });

    const {
        doneEditing,
        currentSelectedPaymentMethod,
        isEditModalHidden,
        showEditModal,
        hideEditModal,
        handlePaymentError,
        handlePaymentSuccess
    } = talonProps;

    const paymentInformation = doneEditing ? (
        <Summary onEdit={showEditModal} isMobile={isMobile} />
    ) : (
        <PaymentMethods
            reviewOrderButtonClicked={reviewOrderButtonClicked}
            selectedPaymentMethod={currentSelectedPaymentMethod}
            onPaymentSuccess={handlePaymentSuccess}
            onPaymentError={handlePaymentError}
            resetReviewOrderButtonClicked={resetReviewOrderButtonClicked}
        />
    );

    const editModal = !isEditModalHidden ? (
        <EditModal onClose={hideEditModal} />
    ) : null;

    return (
        <div className={classes.root}>
            <div className={classes.payment_info_container}>
                {paymentInformation}
            </div>
            {editModal}
        </div>
    );
};

export default PaymentInformation;

PaymentInformation.propTypes = {
    classes: shape({
        container: string,
        payment_info_container: string,
        review_order_button: string
    }),
    reviewOrderButtonClicked: bool,
    isMobile: bool,
    onSave: func.isRequired,
    resetReviewOrderButtonClicked: func.isRequired
};
