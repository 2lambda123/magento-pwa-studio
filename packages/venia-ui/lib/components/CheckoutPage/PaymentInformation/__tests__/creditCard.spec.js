import React from 'react';
import createTestInstance from '@magento/peregrine/lib/util/createTestInstance';
import { useCreditCard } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard';

import CreditCard from '../creditCard';
import LoadingIndicator from '../../../LoadingIndicator';
import Country from '../../../Country';

import classes from '../creditCard.css';

jest.mock('../../../../classify');

jest.mock(
    '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard',
    () => {
        return {
            useCreditCard: jest.fn().mockReturnValue({
                onPaymentError: jest.fn(),
                onPaymentSuccess: jest.fn(),
                onPaymentReady: jest.fn(),
                isBillingAddressSame: false,
                countries: {},
                isLoading: false,
                errors: [],
                stepNumber: 0,
                initialValues: {}
            })
        };
    }
);

jest.mock('../brainTreeDropIn', () => {
    return () => <div>Braintree Dropin Component</div>;
});

jest.mock('../../../LoadingIndicator', () => {
    return () => <div>Loading Indicator Component</div>;
});

jest.mock('../../../Checkbox', () => {
    return props => <div {...props}>Checkout Component</div>;
});

jest.mock('../../../Field', () => {
    return props => <div {...props}>Field Component</div>;
});

jest.mock('../../../TextInput', () => {
    return props => <div {...props}>Text Input Component</div>;
});

jest.mock('../../../Country', () => {
    return props => <div {...props}>Country Component</div>;
});

jest.mock('../../../Region', () => {
    return props => <div {...props}>Region Component</div>;
});

const useCreditCardReturnValue = {
    onPaymentError: jest.fn(),
    onPaymentSuccess: jest.fn(),
    onPaymentReady: jest.fn(),
    isBillingAddressSame: false,
    countries: {},
    isLoading: false,
    errors: [],
    stepNumber: 0,
    initialValues: {
        firstName: 'sample first name',
        lastName: 'sample last name',
        city: 'sample city',
        region: 'sample region',
        country: 'sample country',
        street1: 'sample street 1',
        street2: 'sample street 2',
        postalCode: 'sample postal code',
        phoneNumber: 'sample phone number',
        isBillingAddressSame: false
    },
    shippingAddressCountry: 'US'
};

test('Should return correct shape', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue
    });

    const tree = createTestInstance(<CreditCard />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should render loading indicator if isDoprinLoading is set to true', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue,
        isLoading: true
    });

    const tree = createTestInstance(<CreditCard />);

    expect(tree.root.findByType(LoadingIndicator)).not.toBeNull();
});

test('Should render billing address fields if isBillingAddressSame is false', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue,
        isBillingAddressSame: false
    });

    const tree = createTestInstance(<CreditCard />);

    expect(
        tree.root.findByProps({
            className: classes.billing_address_fields_root
        })
    ).not.toBeNull();
});

test('Billing address fields should not be visibile if isBillingAddressSame is true', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue,
        isBillingAddressSame: true
    });

    const tree = createTestInstance(<CreditCard />);

    expect(
        tree.root.findByProps({
            className: classes.billing_address_fields_root_hidden
        })
    ).not.toBeNull();
});

test('Should render error messages if errors array is not empty', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue,
        errors: ['something is missing']
    });

    const tree = createTestInstance(<CreditCard />);

    expect(
        tree.root.findByProps({ className: classes.errors_container })
    ).not.toBeNull();
});

test('Should use country from shipping address if initialValues is empty', () => {
    useCreditCard.mockReturnValueOnce({
        ...useCreditCardReturnValue,
        isBillingAddressSame: false,
        initialValues: {},
        shippingAddressCountry: 'UK'
    });

    const tree = createTestInstance(<CreditCard />);

    expect(tree.root.findByType(Country).props.initialValue).toBe('UK');
});
