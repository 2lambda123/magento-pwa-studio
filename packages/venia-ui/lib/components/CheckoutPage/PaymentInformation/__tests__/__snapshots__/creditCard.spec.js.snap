// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return correct shape 1`] = `
<div
  className="root"
>
  <div
    className="credit_card_root"
  >
    <div
      className="dropin_root"
    >
      <div>
        Braintree Dropin Component
      </div>
    </div>
    <div
      className="address_check"
    >
      <div
        field="isBillingAddressSame"
        initialValue={false}
        label="Billing address same as shipping address"
      >
        Checkout Component
      </div>
    </div>
    <div
      className="billing_address_fields_root"
    >
      <div
        classes={
          Object {
            "root": "first_name",
          }
        }
        label="First Name"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "last_name",
          }
        }
        label="Last Name"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "country",
          }
        }
        initialValue="sample country"
        validate={[Function]}
      >
        Country Component
      </div>
      <div
        classes={
          Object {
            "root": "street1",
          }
        }
        label="Street Address"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "street2",
          }
        }
        label="Street Address 2"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "city",
          }
        }
        label="City"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "state",
          }
        }
        field="state"
        validate={[Function]}
      >
        Region Component
      </div>
      <div
        classes={
          Object {
            "root": "postal_code",
          }
        }
        label="ZIP / Postal Code"
      >
        Field Component
      </div>
      <div
        classes={
          Object {
            "root": "phone_number",
          }
        }
        label="Phone Number"
      >
        Field Component
      </div>
    </div>
  </div>
</div>
`;
